### PR TITLE
Fix search encoding bug

### PR DESCRIPTION
**Pull request by vsergeenko777**
Fixed issue #5416

### CODE REVIEW

File: src/main/java/appeng/client/gui/me/fluids/FluidRepo.java [hunk 1]
```diff
@@ -38,11 +38,11 @@ public FluidRepo(IScrollSource src, ISortSource sortSrc) {
     @Override
     protected boolean matchesSearch(SearchMode searchMode, Pattern searchPattern, IAEFluidStack stack) {
         if (searchMode == SearchMode.MOD) {
-            String modId = Platform.getModId(stack);
+            String modId = Platform.getModId(stack).toLowerCase();
```
Comment (shartte) [inline src/main/java/appeng/client/gui/me/fluids/FluidRepo.java:41]: This is not necessary if we're matching the pattern case-insensitively.

File: src/main/java/appeng/client/gui/me/fluids/FluidRepo.java [hunk 2]
```diff
@@ -38,11 +38,11 @@ public FluidRepo(IScrollSource src, ISortSource sortSrc) {
...
             return searchPattern.matcher(modId).find();
         }
 
-        String displayName = Platform.getFluidDisplayName(stack).getString();
+        String displayName = Platform.getFluidDisplayName(stack).getString().toLowerCase();
```
Comment (shartte) [inline src/main/java/appeng/client/gui/me/fluids/FluidRepo.java:45]: Same here

File: src/main/java/appeng/client/gui/me/items/ItemRepo.java [hunk 3]
```diff
@@ -43,13 +43,13 @@ public ItemRepo(IScrollSource src, ISortSource sortSrc) {
     @Override
     protected boolean matchesSearch(SearchMode searchMode, Pattern searchPattern, IAEItemStack stack) {
         if (searchMode == SearchMode.MOD) {
-            String modId = Platform.getModId(stack);
+            String modId = Platform.getModId(stack).toLowerCase();
```
Comment (shartte) [inline src/main/java/appeng/client/gui/me/items/ItemRepo.java:46]: Same here

File: src/main/java/appeng/client/gui/me/items/ItemRepo.java [hunk 4]
```diff
@@ -43,13 +43,13 @@ public ItemRepo(IScrollSource src, ISortSource sortSrc) {
...
             return searchPattern.matcher(modId).find();
         }
 
         AEItemStack aeStack = (AEItemStack) stack;
 
-        String displayName = aeStack.getDisplayName().getString();
+        String displayName = aeStack.getDisplayName().getString().toLowerCase();
```
Comment (shartte) [inline src/main/java/appeng/client/gui/me/items/ItemRepo.java:52]: See above

File: src/main/java/appeng/client/gui/me/items/ItemRepo.java [hunk 5]
```diff
@@ -58,7 +58,7 @@ protected boolean matchesSearch(SearchMode searchMode, Pattern searchPattern, IA
             List<Component> tooltip = aeStack.getToolTip();
 
             for (Component line : tooltip) {
-                if (searchPattern.matcher(line.getString()).find()) {
+                if (searchPattern.matcher(line.getString().toLowerCase()).find()) {
```
Comment (shartte) [inline src/main/java/appeng/client/gui/me/items/ItemRepo.java:61]: See above

**Review by shartte — APPROVED** [review]

**Review by Technici4n — APPROVED** [review]

**Pull request merged**